src/Chapter3.hs

```diff
@@ -971,6 +1102,25 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Enum, Eq, Show)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: DayOfWeek -> Int
+daysToParty day = go 1 (nextDay day)
+  where 
+    go :: Int -> DayOfWeek -> Int
+    go acc dd = if dd == Friday 
+      then acc
+      else go (acc + 1) (nextDay dd)
```

Comment:
That's nice recursive implementation! Though, it is possible to implement in a single line, knowing how `mod` works for negative numbers :slightly_smiling_face: 

```haskell
ghci> mod (-1) 7
6
```